state/factory/workingset.go

```diff
@@ -166,6 +166,26 @@ func (ws *workingSet) runAction(
 	if !ok {
 		return nil, errors.New("protocol is empty")
 	}
+	// if it's an EVM tx container, unfold the tx inside
+	if container, ok := selp.Action().(action.EvmTxContainer); ok {
+		var (
+			isContract bool
+			err        error
+		)
+		if container.To() == nil {
+			isContract = true
+		} else {
+			ioAddr, _ := address.FromBytes(container.To().Bytes())
+			isContract, err = ws.checkContract(ctx, ioAddr)
+		}
```

Comment:
code may be clear as:
`isContract, err := ws.checkContract(ctx, container.To())`

Reply:
updated